content/articles/how-to-use-brotli-compression-method/index.md

```diff
@@ -0,0 +1,112 @@
+### Introduction
+Static assets such as *.js*, *.CSS* and *.json* text files are commonly used in web application projects, with redundancy and compression playing an important role. ,Data compression is the process of encoding files to reduce their size, making them take up less space on your server and allowing them to be transferred to site clients more quickly. Compression reduces the amount of bandwith required for website rendering, which speeds up the process.
+
+### Prerequisites
+- Compression algorithm
+
+### What is Brotli
+Brotli is a lossless data compression algorithm that employs a variant of the *LZ77 algorithm*, as well as *Huffman coding* and *second-order context modelling*. Its compression ratio is comparable to the best general-purpose compression methods currently available.
+
+### How Brotli works
+The browser sends an `Accept-Encoding` header with the algorithms it supports and their priority order; the server chooses one, uses it to compress the body of the `content-encoding` header to inform the browser of its choice. Because content negotiation was used to select a representation based on its encoding, the server must include a vary header in the response that includes at least `Accept-encoding`; this allows caches to cache the various representations of the resource.

```

Comment:
`Accept-Encoding` and `content-encoding`

What is this for? Add example image or sample header to explain them first.

Reply:
` it supports and their priority order;`

What do you mean by priority order?

Reply:
`content negotiation was used to select a representation based on its encoding`

Explain

Reply:
`caches to cache the various representations of the resource.`

Explain